client/app/authenticators/zap-api-authenticator.js

```diff
@@ -0,0 +1,67 @@
+import BaseAuthenticator from 'ember-simple-auth/authenticators/base';
+import fetch from 'fetch';
+import location from 'ember-simple-auth/utils/location';
+import { inject as service } from '@ember/service';
+import ENV from 'labs-zap-search/config/environment';
+
+// lifted from https://github.com/simplabs/ember-simple-auth/blob/master/addon/mixins/oauth2-implicit-grant-callback-route-mixin.js#L6
+// parses a window hash and grabs the access token
+function _parseResponse(locationHash) {
+  const params = {};
+  const query = locationHash.substring(locationHash.indexOf('?'));
+  const regex = /([^#?&=]+)=([^&]*)/g;
+  let match;
+
+  // decode all parameter pairs
+  while ((match = regex.exec(query)) !== null) { // eslint-disable-line
+    params[decodeURIComponent(match[1])] = decodeURIComponent(match[2]);
+  }
+
+  return params;
+}
+
+export default class ZAPAuthenticator extends BaseAuthenticator {
+  @service
+  store
+
+  // Authentication is implicit with an httponly cookie holding
+  // a ZAP Token. This request will fail if that cookie doesn't exist
+  // https://github.com/simplabs/ember-simple-auth/blob/master/guides/managing-current-user.md#using-a-dedicated-endpoint
+  async _fetchUserObject() {
+    const user = await this.store.queryRecord('user', { me: true });
+
+    return { id: user.id, ...user.toJSON() };
+  }
+
+  async authenticate() {
+    // Extract NYCID Token from URL
+    const { NYCIDToken } = _parseResponse(location().hash);
```

Comment:
I think there are some missing commits from another branch. `NYCIDToken` is the wrong name — it's called `access_token`